src/main/scala/strawman/collection/Map.scala

```diff
@@ -8,16 +8,16 @@ import scala.annotation.unchecked.uncheckedVariance
 import scala.util.hashing.MurmurHash3
 
 /** Base Map type */
-trait Map[K, +V] extends Iterable[(K, V)] with MapOps[K, V, Map, Map[K, V]]
+trait Map[K, +V] extends Iterable[(K, V)] with MapOps[K, V, Map, Map[K, V]] {
+  protected[this] def c: this.type = this
```

Comment:
I provide a default implementation of `c` in the `Map` branch (and also `Set` and `Seq`, see below). The implementation of `c` differs only in the case of decorators (such as `StringOps` and `ArrayOps` -- and, more generally, if we want to be able to decorate any Java collection we will be able to use the same pattern).